File: backend/src/main/java/com/nekobyte/englishtek/model/Badge.java
package com.nekobyte.englishtek.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import java.util.HashSet;
import java.util.Set;

@Data
@Entity
@Table(name = "badges")
@EqualsAndHashCode(exclude = "userBadges")
@ToString(exclude = "userBadges")
public class Badge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String name;

    @NotBlank
    private String description;

    @NotBlank
    private String iconUrl;

    @OneToMany(mappedBy = "badge", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<UserBadge> userBadges = new HashSet<>();
}

File: backend/app/src/main/java/com/example/englishtek_mobile/data/model/Badge.kt
package com.example.englishtek_mobile.data.model

import com.google.gson.annotations.SerializedName

data class Badge(
    val id: String,
    val name: String?,
    val description: String?,
    @SerializedName("level")
    val level: String?,
    @SerializedName("iconUrl")
    val iconUrl: String?,
    @SerializedName("dateObtained")
    val dateObtained: String? = null
)
